### PR TITLE
wallet_keypool_topup.py: Test for all keypool address types

### PR DESCRIPTION
**Pull request by instagibbs**
To protect against regressions if key scanning is changed.

### CODE REVIEW

File: test/functional/wallet_keypool_topup.py [hunk 1]
```diff
@@ -40,32 +40,37 @@ def run_test(self):
...
         connect_nodes_bi(self.nodes, 0, 1)
+        connect_nodes_bi(self.nodes, 0, 2)
+        connect_nodes_bi(self.nodes, 0, 3)
 
-        self.log.info("Generate keys for wallet")
-        for _ in range(90):
-            addr_oldpool = self.nodes[1].getnewaddress()
-        for _ in range(20):
-            addr_extpool = self.nodes[1].getnewaddress()
+        for i, output_type in enumerate(["legacy" , "p2sh-segwit", "bech32"]):
```
Comment (practicalswift) [inline test/functional/wallet_keypool_topup.py]: Nit: Drop space before "," :-)

File: test/functional/wallet_keypool_topup.py [hunk 2]
```diff
@@ -40,32 +40,37 @@ def run_test(self):
...
 
-        self.log.info("Send funds to wallet")
-        self.nodes[0].sendtoaddress(addr_oldpool, 10)
-        self.nodes[0].generate(1)
-        self.nodes[0].sendtoaddress(addr_extpool, 5)
-        self.nodes[0].generate(1)
-        sync_blocks(self.nodes)
+            self.log.info("Generate keys for wallet with address type: " + output_type)
```
Comment (practicalswift) [inline test/functional/wallet_keypool_topup.py]: Make this logging lazy by using string format arguments :-)

**Review by promag — COMMENTED** [review]
utACK.

File: test/functional/wallet_keypool_topup.py [hunk 3]
```diff
@@ -40,32 +40,37 @@ def run_test(self):
...
+            idx = i+1
+            for _ in range(90):
+                addr_oldpool = self.nodes[idx].getnewaddress(output_type)
+            for _ in range(20):
+                addr_extpool = self.nodes[idx].getnewaddress(output_type)
```
Comment (MarcoFalke) [inline test/functional/wallet_keypool_topup.py]: Why are you passing output_type to be the label of the new address?

**Review by maflcko — DISMISSED** [review]
The test is testing nothing

File: test/functional/wallet_keypool_topup.py (hunk 3 above)
Comment (instagibbs) [reply test/functional/wallet_keypool_topup.py]: because this test isn't actually testing anything. Thanks. Will fix.

Comment (instagibbs) [reply test/functional/wallet_keypool_topup.py]: done

File: test/functional/wallet_keypool_topup.py (hunk 2 above)
Comment (instagibbs) [reply test/functional/wallet_keypool_topup.py]: done

File: test/functional/wallet_keypool_topup.py (hunk 1 above)
Comment (instagibbs) [reply test/functional/wallet_keypool_topup.py]: done

File: test/functional/wallet_keypool_topup.py (hunk 3 above)
Comment (promag) [reply test/functional/wallet_keypool_topup.py]: wow, should we promote named parameters? IIUC it's missing `-addresstype` in L28?

**Comment by instagibbs** [comment]
Tested that the returned addresses appear to be the expected type.

**Comment by MeshCollider** [comment]
utACK https://github.com/bitcoin/bitcoin/pull/14857/commits/0dcac51049cdd924a50d62629757effc8d542046

**Pull request merged**